src/app/pages/PerpetualPage/contexts/GraphQLProvider.tsx

```diff
@@ -0,0 +1,210 @@
+import React, { useState, useEffect } from 'react';
+import { ApolloClient, InMemoryCache, ApolloProvider } from '@apollo/client';
+
+type GraphQLEndpoint = { graph: string; index?: string; isFallback?: boolean };
+
+const metaQuery = `{
+  _meta{
+    block {
+      hash
+    }
+    deployment
+    hasIndexingErrors
+  }
+}`;
+
+const statusQuery = deployment => `{
+  indexingStatuses(subgraphs: [${JSON.stringify(deployment)}]) {
+    subgraph
+    synced
+    health
+    entityCount
+    fatalError {
+      handler
+      message
+      deterministic
+    }
+    chains {
+      chainHeadBlock {
+        number
+      }
+      earliestBlock {
+        number
+      }
+      latestBlock {
+        number
+      }
+    }
+  }
+}`;
+
+const createClient = (endpoint: GraphQLEndpoint) =>
+  new ApolloClient({
+    uri: endpoint.graph,
+    cache: new InMemoryCache({
+      resultCaching: false,
+    }),
+  });
+
+type EvaluationResult = {
+  score: number;
+  isUp?: boolean;
+  isSynced?: boolean;
+  isHealthy?: boolean;
+};
+
+const evaluateEndpoint = async (endpoint: GraphQLEndpoint) => {
+  const tests = 3;
+  let score = 0;
+
+  let result: EvaluationResult = {
+    score: 0,
+    isUp: false,
+    isSynced: undefined,
+    isHealthy: undefined,
+  };
+
+  try {
+    const metaResponse = await (
+      await fetch(endpoint.graph, {
+        method: 'POST',
+        body: JSON.stringify({ query: metaQuery }),
+      })
+    ).json();
+
+    if (!metaResponse?.data?._meta?.deployment) {
+      console.error('Failed to fetch graph deployment id!', metaResponse);
+      return result;
+    }
+
+    score++;
+    result.isUp = true;
+
+    if (!endpoint.index) {
+      // endpoint can not be evaluated further
+      result.score = score / tests;
```

Comment:
I don't think you need to divide `score` by number of `tests` (you would just need to adjust lines `153` and `179`).

This is just an opinion so I'll leave it up to you if you want to change it but you would get rid of one variable and 2 divisions.

Reply:
I added it, because the original thought was to have percentage scrores, so you could calculate in the exact amount of blocks behind head and compare multiple nodes, but it might not really be needed.